subsys/net/lib/sntp/sntp_pkt.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __SNTP_PKT_H
+#define __SNTP_PKT_H
+
+#include <zephyr/types.h>
+
+struct sntp_pkt {
+#if __BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__
```

Comment:
I'm sorry to break fun, but given that the work has started on fixing the bitfield problem properly: https://github.com/zephyrproject-rtos/zephyr/pull/1492 , I fill obliged to comment here. This is *not* a proper fix to the problem. Nowhere does the C standard say that bitfield allocation order depends on endianness of the platform. What it is (quoting @mnkp from the mailing list):

> C99 §6.7.2.1, paragraph 10 says: "The order of allocation of bit-fields within a unit (high-order to low-order or low-order to high-order) is implementation-defined."

That's it - you can't rely that bitfield order is tied to endianness or anything else. It's simply undefined, that's all.

Note that I'm personally ok to go for half-measures like in this patch (or to postpone a real fix until later). Just let's all of us be aware that we're applying half-measures, not real solutions.

Reply:
@pfalcon Let's get first a compiler used to compile Zephyr that fucks this up. Then maybe let's fix it. This issue is exactly like theory vs practice. To be honest, I am pretty sure we won't face any issue with this. Current compilers in Zephyr are just fine with it (they follow what gcc does, the order is endian dependent then) and I would be really surprised we start to support something exotic that is - stupid enough - to mess this up.

Reply:
@tbursztyka :

> This issue is exactly like theory vs practice. To be honest, I am pretty sure we won't face any issue with this. 

Well, I was pretty sure we won't face any issues with make, and yet it's being replaced with CMake... Anyway, I'm fine with that.

Reply:
Well, cmake does create makefiles so in a way make is not really replaced. It's more like its usage is improved by using cmake.

Reply:
@tbursztyka : Let's hope everything will be good ("improved") then ;-).

You're still -1, on this patch, is anything needs fixing yet?

Reply:
> Let's get first a compiler used to compile Zephyr that fucks this up. Then maybe let's fix it. This issue is exactly like theory vs practice. To be honest, I am pretty sure we won't face any issue with this.

That's a very shortsighted way of thinking. Especially know that we move to CMake and are actually likely to support new compilers in a foreseeable future. In any case there is no good excuse for accepting broken code into the code base. It's best to do things right from the very beginning rather than try to fix them later.

> I would be really surprised we start to support something exotic that is - stupid enough - to mess this up.

Compiler which chooses to use different assignment order for C style bit-fields than gcc does is not stupid. Though it's indeed a shame that the C Standard did not decided to enforce stricter encoding rules.

Reply:
Honestly, I even don't know if BYTE_ORDER works for compilers other than GCC.
Let's prefer portability rather than convenience if we will support other compilers. I will replace bitfield by mask and shift.